src/core_modules/capture-core/metaDataMemoryStoreBuilders/programs/factory/enrollment/EnrollmentFactory.js

```diff
@@ -213,18 +238,52 @@ export class EnrollmentFactory {
                         return acc;
                     }, {});
 
-                // $FlowFixMe
-                cachedProgramSections.asyncForEach(async (programSection) => {
-                    section = await this._buildSection(
-                        programSection.trackedEntityAttributes.map(id => trackedEntityAttributeDictionary[id]),
-                        programSection.displayFormName,
-                        programSection.id,
-                    );
-                    section && enrollmentForm.addSection(section);
-                });
+                if (this.dataEntryFormConfig) {
+                    // $FlowFixMe
+                    this.dataEntryFormConfig.asyncForEach(async (formConfigSection) => {
+                        const attributes = formConfigSection.elements.reduce((acc, element) => {
+                            if (element.type === 'plugin') {
+                                const fieldMap = element
+                                    .fieldMap
+                                    ?.map(field => ({
+                                        ...field,
+                                        ...trackedEntityAttributeDictionary[field.IdFromApp],
+                                    }));
+
+                                acc.push({ ...element, fieldMap });
+                                return acc;
+                            }
+                            const attribute = trackedEntityAttributeDictionary[element.id];
+                            if (attribute) {
+                                acc.push(attribute);
+                            }
+                            return acc;
+                        }, []);
+
+                        section = await this._buildSection(
+                            attributes,
```

Comment:
NIT: should probably create a new flow type definition for `sourceElements` in the `_buildSection` function.